main/accounts/types.ts

```diff
@@ -109,8 +120,28 @@ export interface DefaultSignTypedDataRequest extends AccountRequest<'signTypedDa
   typedMessage: TypedMessage
 }
 
+interface EIP2612PermitDomain {
+  chainId: number
+  verifyingContract: string
+}
+
 export interface PermitSignatureRequest extends AccountRequest<'signErc20Permit'> {
-  typedMessage: TypedMessage
+  typedMessage: {
```

Comment:
can we separate this out into a different type? also are permit and tokendata optional?